api/routes.go

```diff
@@ -103,10 +112,22 @@ func (api *API) withAuth(h httprouter.Handle) httprouter.Handle {
 				api.WriteError(w, err, http.StatusUnauthorized)
 				return
 			}
+			sub, _, _, err := jwt.TokenFields(token)
+			if err != nil {
+				api.staticLogger.Debugln("Error decoding token from request:", err)
+				api.WriteError(w, err, http.StatusUnauthorized)
+				return
+			}
+			u, err = api.staticDB.UserBySub(req.Context(), sub)
+			if err != nil {
+				api.staticLogger.Debugln("Error fetching user by token from request:", err)
+				api.WriteError(w, err, http.StatusUnauthorized)
```

Comment:
should this be a 404?

Reply:
I'd say if we were able to fetch a token and were also able to verify that it's a valid one which we provided ourselves but still can't find it in our database it's a server error. So something 5xx.
There is nothing the user did wrong in that case.

But it doesn't seem we actually validate the token here? We just decode it? Doesn't that mean as long as I know the sub I can create fake JWTs for it?

Reply:
What I had in mind here is that someone might have deleted the user in the mean time. Which has its own error, which I should check for. In the general case I agree with 500.

404 won't work here because this is a middleware that wraps a bunch of endpoints - it would signal to the client that the endpoint doesn't exist, which is not the case.